tests/testsuite/lints/blanket_hint_mostly_unused.rs

```diff
@@ -0,0 +1,160 @@
+use crate::prelude::*;
+use cargo_test_support::project;
+use cargo_test_support::str;
+
+#[cargo_test(nightly, reason = "-Zhint-mostly-unused is unstable")]
+fn named_profile_blanket() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+[package]
+name = "foo"
+version = "0.0.1"
+edition = "2015"
+
+[profile.dev]
+hint-mostly-unused = true
+"#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+    p.cargo("check -Zprofile-hint-mostly-unused -v")
+        .masquerade_as_nightly_cargo(&["profile-hint-mostly-unused", "cargo-lints"])
+        .with_stderr_data(str![[r#"
+[WARNING] `hint-mostly-unused` is being blanket applied to all dependencies
+ --> Cargo.toml:8:1
+  |
+7 | [profile.dev]
+8 | hint-mostly-unused = true
+  | ^^^^^^^^^^^^^^^^^^^^^^^^^
+  |
+  = [NOTE] `cargo::blanket_hint_mostly_unused` is set to `warn` by default
+[HELP] try enabling `hint-mostly-unused` for just one package
```

Comment:
I was concerned with the `try` language but it seems to be common in rustc.  However, in writing up an alternative, I think I like that better

> you can scope `hint-mostly-unused` to specific packages

maybe even better if say we

> scope `hint-mostly-unused` to specific packages with a lot of unused object code

We may even want to say that on all of them, even if we don't suggest the fix